chain/indexer/src/block.rs

```diff
@@ -141,168 +135,373 @@ where
-                                checksum = %current_described_block.checksum,
-                                "Indexer state update",
-                            );
-
-                            #[cfg(all(feature = "prometheus", not(test)))]
-                            {
-                                let low_4_bytes = hopr_primitive_types::prelude::U256::from_big_endian(
-                                    current_described_block.checksum.as_ref(),
-                                )
-                                .low_u32();
-                                METRIC_INDEXER_CHECKSUM.set(low_4_bytes.into());
+                .filter_map(|block| {
+                    let db = db.clone();
+                    let db_processor = db_processor.clone();
+
+                    async move {
+                        match Self::process_block_by_id(&db, &db_processor, block.block_id).await {
+                            Ok(events) => events,
+                            Err(e) => {
+                                error!("Failed to process logs from {block}: {e}");
+                                None
                             }
                         }
-                        Err(e) => error!(error = %e, "Cannot retrieve indexer state"),
                     }
-
-                    outgoing_events
                 })
                 .flat_map(stream::iter);
 
-                futures::pin_mut!(event_stream);
-                while let Some(event) = event_stream.next().await {
-                    trace!(event=%event, "Processing onchain event");
-                    // Pass the events further only once we're fully synced
-                    if is_synced.load(std::sync::atomic::Ordering::Relaxed) {
-                        if let Err(e) = tx_significant_events.try_send(event) {
-                            error!(error = %e,"failed to pass a significant chain event further");
-                        }
+            futures::pin_mut!(event_stream);
+            while let Some(event) = event_stream.next().await {
+                trace!(event=%event, "Processing onchain event");
+                // Pass the events further only once we're fully synced
+                if is_synced.load(Ordering::Relaxed) {
+                    if let Err(e) = tx_significant_events.try_send(event) {
+                        error!(error = %e,"failed to pass a significant chain event further");
                     }
                 }
+            }
 
+            if panic_on_completion {
                 panic!(
                     "Indexer event stream has been terminated, cannot proceed further!\n\
-                This error indicates that an issue has occurred at the RPC provider!\n\
-                The node cannot function without a good RPC connection."
+                    This error indicates that an issue has occurred at the RPC provider!\n\
+                    The node cannot function without a good RPC connection."
```

Comment:
@ausias-armesto would want this to be a single line though. Can you pls simplify into a single line?